README.md
# Winfield child Theme

## Installation

1. Upload the Winfield child theme folder via FTP to your wp-content/themes/ directory. (The Genesis parent theme needs to be in the wp-content/themes/ directory as well.)

2. Go to your WordPress dashboard and select Appearance.

3. Activate the Winfield child theme.

4. Inside your WordPress dashboard, go to Genesis > Theme Settings and configure them to your liking.

## Note

After you download and unarchive the theme from GitHub, rename the Winfield-master folder to Winfield.

## Widgets

The homepage features 5 custom widget areas:

1. Home Slider. For use with the Genesis Responsive Slider plugin.

2. Home Welcome. Use a text widget to display a brief message.

3. Home Left. For use with the Genesis Featured Page widget. May substitute Genesis Featured Posts widget.

4. Home Center. For use with the Genesis Featured Page widget. May substitute Genesis Featured Posts widget.

5. Home Right. For use with the Genesis Featured Page widget. May substitute Genesis Featured Posts widget.

An aditional widget area called After Post Block appears after a post on all single post pages. Use a text widget or even the Genesis eNews Extended widget here.

## Widget Page Template

Winfield comes with a widget Page Template. To use it, publish an empty page, select the Widget Page Template from the drop down menu then start adding widgets
to it. Refer to the Demo site for an example.

## Configuring the widgets as seen on the Demo site

Genesis Responsive Slider:

1. Maximum Slider Width = 1180.

2. Maximum Slider Height = 400.

3. Check Display Next / Previous Arrows in Slider?

4. Check Display Pagination in Slider?

5. Check Display Post/Page Title in Slider?

6. Check Display Content in Slider?

7. Select Display post content from drop menu.

8. Enter 150 into Limit content to text field.

9. Slider Excerpt Width = 75.

10. Excerpt Location vertical = Bottom.

11. Excerpt Location horizontal = Left.


Home Left / Home Center / Home Right - Genesis Featured Page

1. Check Show Featured Image.

2. Choose home-middle (340 x 150) for Image Size.

3. Choose None for Image Alignment.

4. Check Show Post Title.

5. Check Show Page Content

6. Select Show Content Limit from drop down menu and enter number of characters. Demo site uses 150.

If using Genesis Featured Posts widget here:

1. Select Category from drop down menu.

2. Choose 1 for Number of Posts to Show.

3. Check Show Featured Image.

4. Choose either square(120x120) thumbnail(150x150) or mini(80x80).

5. Choose Left or Right for Image Alignment.

6. Check Show Post Title. Do not check Show Post Info.

## Responsive Menus

Winfield comes with built-in responsive support for the Primary, Secondary and Header navigation menus. They will kick in at the 768px breakpoint.

IMPORTANT!! Make sure to call your menus Primary Navigation, Secondary Navigation and Header Navigation as those are the names the javascript file is looking for. It doesn't matter where you place them but they have to use those names.

While this child theme is internationalized if you wish to translate the output of the responsive menus you'll need to do so manually. This is because the output is not generated from the javascript file but rather from the style sheet using the content property. I currently know of no way to automatically translate CSS generated content.

At the 768px breakpoint you'll see that the menu output looks like this => HEADER MENU, PRIMARY MENU, SECONDARY MENU. To change this output open the child theme's style.css file and scroll toward the bottom. The section you are looking for is this ...

<code>/*
18 Media Queries
---------------------------------------------------- */</code>

Continue scrolling down a bit until you find this ...

<code>/* Menu Icon properties */</code>

There are 3 blocks that will need to be manually changed.

FIRST BLOCK:

	#header-menu-icon::before {
		content: "HEADER MENU \25BC";
	}

	#primary-menu-icon::before {
		content: 'PRIMARY MENU \25BC';
	}

	#secondary-menu-icon::before {
		content: 'SECONDARY MENU \25BC';
	}
	
SECOND BLOCK:

	#header-menu-icon:hover::before {
		content: "HEADER MENU \25BC";
	}

	#primary-menu-icon:hover::before {
		content: 'PRIMARY MENU \25BC';
	}

	#secondary-menu-icon:hover::before {
		content: 'SECONDARY MENU \25BC';
	}
	
THIRD BLOCK:

	#header-menu-icon:active::before {
		content: "HEADER MENU \25BC";
	}

	#primary-menu-icon:active::before {
		content: 'PRIMARY MENU \25BC';
	}

	#secondary-menu-icon:active::before {
		content: 'SECONDARY MENU \25BC';
	}
	
The First Block deals with the output of the menus normal state, the Second Block deals with the output of the menus hover state and the Third Block deals with the output of the menus active (or onclick) state.

This is CSS generated content. Using header-menu for example, you see HEADER MENU on a mobile or when you re-size your desktop browser window because of this line ... <code>content: "HEADER MENU \25BC";</code>

You may change HEADER MENU to whatever you want. The 25BC is a unicode which translates to an upside down triangle. Change the terms HEADER MENU, PRIMARY MENU and SECONDARY MENU to whatever you'd like. Do so in ALL 3 instances as referenced above.

Let's use a real life example. Let's assume you operate a Dutch language website. You've translated the child theme using the included .POT file. Now you want to change the phrase PRIMARY MENU. Swap out PRIMARY MENU for Hoofdmenu. It should now look like this ...

	#header-menu-icon::before {
	content: "Hoofdmenu \25BC";
	}
	
Do so for all 3 instances.

## Content Boxes:

The custom content boxes make use of DIV classes. For example ...

<code>&lt;div class="content-box-blue"&gt;some random text&lt;/div&gt;</code>

There are 7 styles to choose from. Refer to the style sheet for details.

## Gradient Buttons:

Each button has a unique class. For example ...

<code>&lt;a class="button-green" href="#"&gt;Link Text&lt;/a&gt;</code> 

The above will render the Green button. There are 11 styles to choose from. Refer to the style sheet for details.

## Recommended Plugins

1. [Genesis Responsive Slider](http://wordpress.org/extend/plugins/genesis-responsive-slider)

2. [Genesis eNews Extended](http://wordpress.org/extend/plugins/genesis-enews-extended/)

3. [Genesis Widgetized Archive](http://wordpress.org/plugins/genesis-widgetized-archive/)

4. [Genesis Widgetized Not Found & 404](http://wordpress.org/plugins/genesis-widgetized-notfound/)

5. [Genesis Box](http://wordpress.org/plugins/wp-genesis-box/)

6. [Regenerate Thumbnails](http://wordpress.org/extend/plugins/regenerate-thumbnails/)

## Credit

1. [Content Boxes by Brian Gardner](http://www.briangardner.com/genesis-content-boxes/)

2. [Gradient Buttons by Brian Gardner](http://www.briangardner.com/social-media-buttons/)

3. [Icons for eNews and Search input fields by iconmonstr](http://iconmonstr.com/)

4. [Responsive Menus by Brad Potter](http://bradpotter.com/)


## Further

1. [Theme Release Page](http://wpcanada.ca/our-themes/winfield/)

2. [Theme Demo](http://demo.wpcanada.ca/winfield/)

3. Download from [Google](http://code.google.com/p/winfield-theme/)

4. Download from [GitHub](https://github.com/lenkutchma/Winfield)

## Bug Reports

1. [Google](http://code.google.com/p/winfield-theme/issues/list)

2. [GitHub](https://github.com/lenkutchma/Winfield/issues)

## Support

The Winfield child theme is offered without support and is intended for users who are knowledgeable with WordPress and the Genesis Framework.

Enjoy!

changelog.txt
*** Winfield Changelog ***


2011.01.20 - Version 1.0

  * changelog.txt - Added changelog and revised version number

2012.01.21 - version 1.0.1
  * updated Genesis Template version number
  * updated theme version number
  * updated changelog

2012.01.21 - version 1.1.0
  * added page-portfolio.php
  * updated style.css and theme version number
  * updated functions.php
  * updated README
  * updated changelog

2012.01.22 - version 1.1.1
  * fixed typo in functions.php
  * fixed typo in page-portfolio.php
  * updated version number
  * updated changelog

2012.01.23 - version 1.2.0
  * removed forced homepage layout from functions.php
  * rewrote home.php and added forced layout as a conditional - if widgets are active full width content in effect otherwise default layout in effect
  * updated version number
  * updated changelog

2012.02.06 - version 1.3.0
  * added Genesis Box
  * added support for Post Formats
  * updated changelog and version number

2012.02.09 - version 1.4.0
  * added internationalization support
  * updated changelog and version number

2012.07.07 - version 1.5.0
  * cleaned up style.css in accordance with Genesis CSS Best Practices == http://www.briangardner.com/code/css-best-practices/
  * restyled Alert Boxes and renamed them Content Boxes. I prefer Brian's naming convention == http://www.briangardner.com/genesis-content-boxes/
  * updated changelog
  * updated README

2012.09.06 - version 2.0
  * complete theme rewrite
  * responsive layout
  * updated style.css
  * updated functions.php
  * updated README
  * updated changelog

2012.09.15 - version 2.0.1
  * fixed bug in home.php that prevented Genesis Responsive Slider from appearing unless another widget was also active
  * updated style.css
  * updated README
  * updated changelog

2012.09.28 - version 2.0.2
  * fixed responsive styling for Genesis Responsive Slider

2012.10.15 - version 2.0.3
  * added separate class for pullquotes and adjusted styling

2012.12.08 - version 2.0.4
  * fixed bug in numerical page navigation

2012.12.15 - version 2.1.0
  * added styling for Page Links (paginated posts)

2012.12.20 - version 2.1.1
  * replaced deprecated add_custom_background() with add_theme_support( 'custom-background' )

2013.01.13 - version 2.2.0
  * added styling for Genesis eNews Extended plugin

03 AUG 2013 - version 2.2.1
  * fixed translation problem

30 SEPT 2013 - version 3.0.0
	* updated theme for HTML5
	* added new widget Page Template
	* updated styling
	* optimized images
	* removed support for Post Formats
	* removed widget support for 404 Error Page in favour of plugin
	* removed support for Genesis Box in favour of plugin
	* introduced new method for responsive menus

02 OCT 2013 - version 3.0.1
	* replaced custom single post navigation with native Genesis single post navigation which was introduced in Genesis 2.0
	* updated functions.php and style.css in keeping up with latest Genesis coding standards

09 OCT 2013 - version 3.0.2
	* added landing Page Template
	* changed hyphen to underscore for widget Page Template name in keeping with WordPress naming standards for specialized template files

18 OCT 2013 - version 3.0.3
	* updated README files

404.php
<?php
/**
 * 404 Not found page example
 * 
 * This helps create a custom and widgetized 404 error page template for Genesis child themes.
 *
 * @author David Decker, http://genesisthemes.de/en/
 * @link http://genesisthemes.de/en/2011-08/tutorial-widgetized-404-error-page-in-genesis/
 *
 * Note by Len, added class 'entry-content' so that styling for 404 page is inherited by theme.
 */


remove_action( 'genesis_loop', 'genesis_do_loop' );
add_action( 'genesis_loop', 'winfield_404_loop_helper' );
/**
 * Add widget support for the 404 page. If no widgets active, display the default loop/message.
 *
 */
function winfield_404_loop_helper() {

	if ( is_active_sidebar( '404-page' ) ) {
			echo '<div class="entry-content">';
		dynamic_sidebar( '404-page' );
			echo '</div><!-- end .entry-content -->';		
	}
	else {
		genesis_standard_loop();
	}
	
}

genesis();

home.php
<?php
/**
 * Setup the home page
*/

add_action( 'genesis_meta', 'winfield_home_genesis_meta' );
/**
 * Add widget support for homepage. If no widgets active, display the default loop.
 *
 */
function winfield_home_genesis_meta() {

	if ( is_active_sidebar( 'home-slider' ) || is_active_sidebar( 'home-welcome' ) || is_active_sidebar( 'home-left' ) || is_active_sidebar( 'home-center' ) || is_active_sidebar( 'home-right' ) ) {

		// Add custom body class to home page
		add_filter( 'body_class', 'winfield_hp_add_body_class' );
		function winfield_hp_add_body_class( $classes ) {
   			$classes[] = 'winfield';
  			return $classes;
		}

		// Remove the default Genesis loop
		remove_action( 'genesis_loop', 'genesis_do_loop' );

		// Force full width content layout
		add_filter( 'genesis_pre_get_option_site_layout', '__genesis_return_full_width_content' );

		// Add homepage widgets
		add_action( 'genesis_loop', 'winfield_homepage_widgets' );

	}
}

function winfield_homepage_widgets() {

	genesis_widget_area( 'home-slider', array(
		'before'	=> '<div class="home-slider widget-area">',
		'after'		=> '</div>',
	) );

	genesis_widget_area( 'home-welcome', array(
		'before'	=> '<div class="home-welcome widget-area">',
		'after'		=> '</div>',
	) );

	if ( is_active_sidebar( 'home-left' ) ||  is_active_sidebar( 'home-center' ) || is_active_sidebar( 'home-right' ) ) {

		echo '<div class="home-middle">';

		genesis_widget_area( 'home-left', array(
			'before'	=> '<div class="home-left widget-area">',
			'after'		=> '</div>',
		) );

		genesis_widget_area( 'home-center', array(
			'before'	=> '<div class="home-center widget-area">',
			'after'		=> '</div>',
		) );

		genesis_widget_area( 'home-right', array(
			'before'	=> '<div class="home-right widget-area">',
			'after'		=> '</div>',
		) );

		echo '</div>';

	}

}

genesis();

page_landing.php
<?php
/**
 * Template Name: Landing Page
 *
 */

//* Add custom body class to landing page
add_filter( 'body_class', 'winfield_lp_add_body_class' );
function winfield_lp_add_body_class( $classes ) {
	$classes[] = 'winfield-landing-page';
	return $classes;

}

//* Force full width content layout
add_filter( 'genesis_pre_get_option_site_layout', '__genesis_return_full_width_content' );

//* Remove site header elements
remove_action( 'genesis_header', 'genesis_header_markup_open', 5 );
remove_action( 'genesis_header', 'genesis_do_header' );
remove_action( 'genesis_header', 'genesis_header_markup_close', 15 );

//* Remove navigation
remove_action( 'genesis_after_header', 'genesis_do_nav' );
remove_action( 'genesis_after_header', 'genesis_do_subnav' );

//* Remove breadcrumbs
remove_action( 'genesis_after_header', 'genesis_do_breadcrumbs' );

//* Remove site footer widgets
remove_action( 'genesis_before_footer', 'genesis_footer_widget_areas' );

//* Remove site footer elements
remove_action( 'genesis_footer', 'genesis_footer_markup_open', 5 );
remove_action( 'genesis_footer', 'genesis_do_footer' );
remove_action( 'genesis_footer', 'genesis_footer_markup_close', 15 );

genesis();

functions.php
<?php
//* Start the engine
include_once( get_template_directory() . '/lib/init.php' );

//* Set Localization (do not remove)
load_child_theme_textdomain( 'winfield', apply_filters( 'child_theme_textdomain', get_stylesheet_directory() . '/lib/languages', 'winfield' ) );

//* Child theme (do not remove)
define( 'CHILD_THEME_NAME', __( 'Winfield Theme', 'winfield' ) );
define( 'CHILD_THEME_URL', 'http://wpcanada.ca/our-themes/winfield' );
define( 'CHILD_THEME_VERSION', '3.0.3' );

//* Add HTML5 markup structure
add_theme_support( 'html5' );

//* Add viewport meta tag for mobile browsers
add_theme_support( 'genesis-responsive-viewport' );

//* Enqueue Google fonts
add_action( 'wp_enqueue_scripts', 'winfield_google_fonts' );
function winfield_google_fonts() {
	wp_enqueue_style( 'google-font', '//fonts.googleapis.com/css?family=Roboto:300,400|Roboto+Slab:300,400', array(), CHILD_THEME_VERSION );
}

/**
 * Register and Enqueue Responsive Menu Script
 * 
 * @author Brad Potter
 * 
 * @link http://www.bradpotter.com
 */
function winfield_menu_script() {
	wp_register_script( 'responsive-menu', get_stylesheet_directory_uri() . '/lib/js/responsivemenus.js', array('jquery'), '1.0.0', false );
	wp_enqueue_script( 'responsive-menu' );

 }
add_action('wp_enqueue_scripts', 'winfield_menu_script');

//* Add new image sizes
add_image_size( 'mini', 80, 80, TRUE );
add_image_size( 'square', 120, 120, TRUE );
add_image_size( 'home-middle', 340, 150, TRUE );
add_image_size( 'slider', 1180, 400, TRUE );

//* Add support for custom background
add_theme_support( 'custom-background' );

//* Add support for custom header
add_theme_support( 'custom-header', array(
	'width' => 1180,
	'height' => 200,
	'header-text'     => false
) );

//* Add support for 3-column footer widgets
add_theme_support( 'genesis-footer-widgets', 3 );

//* Add after post block section to single post page
add_action( 'genesis_entry_footer', 'winfield_after_post_block'  ); 
function winfield_after_post_block() {

    if ( ! is_singular( 'post' ) )
    	return;

    genesis_widget_area( 'after-post-block', array(
		'before' => '<div class="after-post-block widget-area"><div class="wrap">',
		'after'  => '</div></div>',
    ) );

}

/** Reposition the breadcrumbs */
remove_action( 'genesis_before_loop', 'genesis_do_breadcrumbs' );
add_action( 'genesis_after_header', 'genesis_do_breadcrumbs' );

//* Add single post navigation
add_action( 'genesis_before_comments', 'genesis_prev_next_post_nav' );

//* Register widget areas
genesis_register_sidebar( array(
	'id'		=> 'home-slider',
	'name'		=> __( 'Home Slider', 'winfield' ),
	'description'	=> __( 'This is the home slider section.', 'winfield' ),
) );
genesis_register_sidebar( array(
	'id'		=> 'home-welcome',
	'name'		=> __( 'Home Welcome', 'winfield' ),
	'description'	=> __( 'This is the home welcome section.', 'winfield' ),
) );
genesis_register_sidebar( array(
	'id'		=> 'home-left',
	'name'		=> __( 'Home Left', 'winfield' ),
	'description'	=> __( 'This is the home left section.', 'winfield' ),
) );
genesis_register_sidebar( array(
	'id'		=> 'home-center',
	'name'		=> __( 'Home Center', 'winfield' ),
	'description'	=> __( 'This is the home center section.', 'winfield' ),
) );
genesis_register_sidebar( array(
	'id'		=> 'home-right',
	'name'		=> __( 'Home Right', 'winfield' ),
	'description'	=> __( 'This is the home right section.', 'winfield' ),
) );
genesis_register_sidebar( array(
	'id'		=> 'after-post-block',
	'name'		=> __( 'After Post Block', 'winfield' ),
	'description'	=> __( 'This is the after post block section.', 'winfield' ),
) );
genesis_register_sidebar( array(
	'id'		=> 'widget-page',
	'name'		=> __( 'Widget Page', 'winfield' ),
	'description'	=> __( 'This is the Widget Page template.', 'winfield' ),
) );